AABB_tree/include/CGAL/AABB_tree.h

```diff
@@ -670,25 +710,51 @@ namespace CGAL {
     m_need_build = true;
   }
 
+  template<typename Tr>
+  template<typename ConstPrimitiveIterator>
+  void
+  AABB_tree<Tr>::expand(Node& node,
+                        ConstPrimitiveIterator first,
+                        ConstPrimitiveIterator beyond,
+                        const std::size_t range,
+                        const Tr& traits)
+  {
+    node.set_bbox(traits.compute_bbox_object()(first, beyond));
+
+    // sort primitives along longest axis aabb
+    traits.split_primitives_object()(first, beyond, node.bbox());
+
+    switch(range)
+    {
+    case 2:
+      node.set_children(*first, *(first+1));
+      break;
+    case 3:
+      node.set_children(*first, new_node());
+      expand(node.right_child(), first+1, beyond, 2, traits);
+      break;
+    default:
+      const std::size_t new_range = range/2;
+      node.set_children(new_node(), new_node());
+      expand(node.left_child(), first, first + new_range, new_range, traits);
+      expand(node.right_child(), first + new_range, beyond, range - new_range, traits);
+    }
+  }
+
+
 	// Build the data structure, after calls to insert(..)
 	template<typename Tr>
 	void AABB_tree<Tr>::build()
 	{
     clear_nodes();
+
     if(m_primitives.size() > 1) {
 
 			// allocates tree nodes
-			m_p_root_node = new Node[m_primitives.size()-1]();
-			if(m_p_root_node == nullptr)
```

Comment:
If `new` failed, it will throw an exception (even with `-fno-exceptions` unless libc++ is compiled with the flag, which in this case will call abort), and will not return nullptr. So the check is superfluous.